Hydropro case study (#320)
xataio/mdx-blog: case-study-hydropro-solutions.mdx

---

> title: "How HydroPro Solutions achieved a 60% reduction in search time using Xata’s Postgres-powered platform"

Author: ```suggestion
title: "How HydroPro Solutions achieved a 60% reduction in search time using the Xata Postgres-powered platform"
```

> title: 'How HydroPro Solutions achieved a 60% reduction in search time using the Xata Postgres-powered platform'

Author: ```suggestion
title: 'HydroPro Solutions achieves a 60% reduction in search time using the Xata Postgres-powered platform'
```

description: 'Discover how HydroPro Solutions reduced search time by 60% and transformed their operations with the powerful Postgres-based serverless database solution from Xata, streamlining water meter installations and improving overall efficiency.'
image:
  src: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/sava-labs-case-study@2x.png
  alt: Case Study
author: Cezzaine Zaher

> date: 09-18-2024

Reviewer: Should be the 19th if publishing on Thursday

Author: Good catch, thank you!

> date: 09-18-2024

Author: ```suggestion
date: 09-19-2024
```

> We needed a PostgreSQL system that would just work and perform well without the hassle of dealing with core database infrastructure. As a single engineer, I didn’t want to spend time managing configurations—Xata’s APIs and search capabilities made it a perfect fit.\
> _Chris Hiester, Lead Application Engineer at HydroPro Solutions_

## Solution

After evaluating several database solutions, Chris chose Xata for its powerful Postgres-based relational database features and ease of use. Xata provided a developer-friendly Postgres platform with robust search functionality and the ability to manage large datasets efficiently.

With Xata, HydroPro was able to:

- **Store and manage vast amounts of relational data:** Including customer records, water meter data, and work orders.
- **Use advanced search capabilities:** Enabling quick and accurate data retrieval for customer service, ensuring no billing errors or discrepancies during meter installations.

> - **Manage file attachments:** Storing photos taken during meter installations and using Google Cloud Vision to read meter serial numbers, which are automatically linked to customer records.

Reviewer: Not the best wording. I'm trying to sneak in that they used image transformations. Any better ideas?

```suggestion
- **Manage file attachments:** Storing photos taken during meter installations and using Google Cloud Vision to read meter serial numbers, which are automatically linked to customer records. Image transformations are then used when delivered in their mobile and web apps.
```

Author: Gotcha. I have tweaked so it reads a little less confusing for people reading this for the first time.

"Manage file attachments:** Storing photos taken during meter installations and using Google Cloud Vision to read meter serial numbers, which are automatically linked to customer records. Xata's image transformation capabilities create smaller thumbnails of the images, allowing field teams using mobile devices to quickly view and manage high-resolution photos in real-time, improving on-site efficiency."

> - **Manage file attachments:** Storing photos taken during meter installations and using Google Cloud Vision to read meter serial numbers, which are automatically linked to customer records.

Author: ```suggestion
- **Manage file attachments:** Storing photos taken during meter installations and using Google Cloud Vision to read meter serial numbers, which are automatically linked to customer records. Xata's image transformation capabilities create smaller thumbnails of the images, allowing field teams using mobile devices to quickly view and manage high-resolution photos in real-time, improving on-site efficiency.
```

## Challenges and the task ahead

As the sole platform engineer at HydroPro Solutions, Chris Hiester faced a significant challenge: building an entire software platform to manage large-scale water meter replacement projects. These projects involved managing data for up to 20,000 water meters in a single city, including customer information, meter readings, work orders, and high-resolution images of the installations.

Chris needed a database solution that could handle vast amounts of relational data while being easy to manage as a single developer. The system also needed to support search capabilities for quickly locating customer records and photos, avoiding costly billing disputes. Additionally, HydroPro needed the platform to integrate smoothly into their existing tech stack, which included Next.js, Vercel, and Google Cloud Vision.

> > We needed a PostgreSQL system that would just work and perform well without the hassle of dealing with core database infrastructure. As a single engineer, I didn’t want to spend time managing configurations—Xata’s APIs and search capabilities made it a perfect fit.\\

Reviewer: ```suggestion
> We needed a PostgreSQL system that would just work and perform well without the hassle of dealing with core database infrastructure. As a single engineer, I didn’t want to spend time managing configurations—Xata’s APIs and search capabilities made it a perfect fit.\
```

I think we only need one backslash, it's showing up in the quote.

![CleanShot 2024-09-19 at 11 15 30@2x](https://github.com/user-attachments/assets/398304a6-f067-46ef-8598-c119bce51fe0)

![HydroPro Solutions Dashboard](/images/case-study-hydropro/hydropro-dashboard.png)

## Results and impact

Since implementing Xata, HydroPro Solutions has seen significant improvements in their operations, particularly in terms of speed and accuracy. The biggest transformation has been in search functionality, which allows internal teams to quickly locate specific addresses and records across thousands of data points.

**Key benefits include:**

- **60% reduction in search time:** Enabling HydroPro’s teams to resolve customer queries faster and ensure accurate data retrieval.
- **Efficient project management:** Field teams can track project progress in real-time, correlating installation speeds with environmental factors like temperature and rainfall, which impact productivity.
- **Improved file handling:** With Xata, HydroPro now stores and manages thousands of photos, making it easy to resolve any customer disputes about water meter readings by referencing high-resolution images.

> > With Xata, we’ve achieved a 60% reduction in search time. Previously, searching for an address would lead to long load times, sometimes with the system freezing up. Now, it’s lightning-fast—start typing an address, and results filter instantly. It has truly been a game changer.\\
> >
> > _Chris Hiester, Lead Application Engineer at HydroPro Solutions_

Reviewer: ```suggestion
> With Xata, we’ve achieved a 60% reduction in search time. Previously, searching for an address would lead to long load times, sometimes with the system freezing up. Now, it’s lightning-fast—start typing an address, and results filter instantly. It has truly been a game changer.\
>
> _Chris Hiester, Lead Application Engineer at HydroPro Solutions_
```

Same as above
![CleanShot 2024-09-19 at 11 16 48@2x](https://github.com/user-attachments/assets/2b71d4d3-095d-499a-b3f1-2cfdaba0da5e)
